torch/functional.py

```diff
@@ -245,3 +246,23 @@ def maybeSqueeze(tensor):
 
     raise ValueError("both arguments to __matmul__ need to be at least 1D, "
                      "but they are {}D and {}D".format(dim_tensor1, dim_tensor2))
+
+
+def sum(input, axes, keepdims=False, out=None):
+    if isinstance(axes, collections.Iterable)
+        if a.dim() > 3:
+            if keepdims:
+                for ax in axes:
+                    input = input.sum(ax, keepdims=True)
+            else:
+                for ax in sorted(axes, reverse=True):
+                    input = input.sum(ax)
```

Comment:
Here size of the tensor changes with each iteration, so it anyway requires to allocate new memory every time. In this case `out` won't do the job unless I don't know about some advanced features of Pytorch memory management...